code/SecureFileExtension.php

```diff
@@ -99,6 +90,35 @@ public function canView($member = null) {
 			return $file->Parent()->canView($member);
 		}
 
+		return $this->defaultPermissions($member);
+	}
+
+	/**
+	 * Checks for any default access permissions and tests against them if found. Default permssions are set via the 
+	 * Config system.
+	 *
+	 * @param Member $member
+	 * @return boolean
+	 */
+	protected function defaultPermissions($member = null) {
```

Comment:
If we change the calls from $this->defaultPermissions() to $this->owner->defaultPermissions() and make it public, then it could be overridden.